llama-index-integrations/indices/llama-index-indices-managed-postgresml/llama_index/indices/managed/postgresml/retriever.py

```diff
@@ -25,61 +25,42 @@ def __init__(
         self,
         index: PostgresMLIndex,
         callback_manager: Optional[CallbackManager] = None,
-        **kwargs: Any,
+        query: Optional[Dict[str, Any]] = None,
+        limit: Optional[int] = 5,
+        rerank: Optional[Dict[str, Any]] = None,
+        **kwargs,
     ) -> None:
         """Initialize params."""
         self._index = index
+        self._query = query
```

Comment:
similar comment here, curious what the use case is for attaching a static query to the class?

Reply:
Probably bad variable choice on my part. This integration is essentially a wrapper around around our actual SDK: https://postgresml.org/docs/api/client-sdk/ 

With our SDK the user has a ton of control over how they want to structure the query. This is basically an escape hatch where if they really want to use that full power they can. Ideally it would be passed directly into the retrieve or aretrieve function, but those don't take kwargs, so this is the best I could come up with. 

It is used here: https://github.com/SilasMarvin/llama_index/blob/a0a0bc215ce04f3daf2de755bf96cafff75094dd/llama-index-integrations/indices/llama-index-indices-managed-postgresml/llama_index/indices/managed/postgresml/retriever.py#L53

If they provide it we completely ignore all other arguments.

To be honest I don't like how I did it, but I'm not sure of a better way to accomplish this. If you have some thoughts for a better integration please let me know!

Reply:
Fair enough! Does a better name help here?

Reply:
I think its mostly the name I found confusing, query vs. QueryBundle vs. Query string

Reply:
I renamed it to `pgml_query` I think that does make a lot more sense. Let me know what you think! https://github.com/run-llama/llama_index/pull/14134/commits/c0eb0b46c8a588e1a7ac4a77585b6b7febabe8a1

Reply:
thank you! That makes much more sense now

Reply:
Of course. Great call!